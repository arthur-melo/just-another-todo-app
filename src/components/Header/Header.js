import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';

const headerPropTypes = {
  className: PropTypes.string,
};

const Header = ({ className }) => (
  <div className={className}>
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon className="d-inline-block" icon={faListAlt} size="lg" alt="List emoji" />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  </div>
);

Header.propTypes = headerPropTypes;

export default Header;
