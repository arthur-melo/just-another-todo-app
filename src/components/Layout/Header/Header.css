.header__container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background: var(--background-color);
}

.header__span {
  margin: 0 0.2em;
  vertical-align: super;
  font-size: 1.3em;
}
