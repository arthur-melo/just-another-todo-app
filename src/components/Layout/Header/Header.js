import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  );
};

export default Header;
