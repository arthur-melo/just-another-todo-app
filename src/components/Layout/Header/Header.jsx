import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="d-flex justify-content-center align-items-center w-100 h-100 bg-light">
      <nav className="navbar">
        <span className="navbar-brand m-0">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__text align-top fs-5">
          Just Another Todo App
        </span>
      </nav>
    </div>
  );
};

export default Header;
