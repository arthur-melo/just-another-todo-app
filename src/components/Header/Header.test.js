import React from 'react';

import { render } from '@testing-library/react';

import Header from './Header';

describe('Header', () => {
  it('should render the Header component', () => {
    const className = 'test-classname';
    const { container } = render(<Header className={className} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
