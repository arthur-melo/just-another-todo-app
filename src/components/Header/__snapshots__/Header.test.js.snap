// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header component 1`] = `
<div
  class="test-classname"
>
  <div
    class="header__container"
  >
    <nav
      class="navbar"
    >
      <span
        class="header__span navbar-brand"
      >
        <svg
          alt="List emoji"
          aria-hidden="true"
          class="svg-inline--fa fa-list-alt fa-w-16 fa-lg d-inline-block"
          data-icon="list-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zM128 120c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm288-136v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="header__span"
      >
        Just Another Todo App
      </span>
    </nav>
  </div>
</div>
`;
